src/libraries/System.Data.OleDb/src/OleDbDataReader.cs

```diff
@@ -1762,7 +1762,7 @@ private void GetRowHandles(/*int skipCount*/)
                 ProcessResults(hr);
             }
             _isRead = ((OleDbHResult.DB_S_ENDOFROWSET != hr) || (0 < (int)_rowFetchedCount));
-            _rowFetchedCount = (IntPtr)Math.Max((int)_rowFetchedCount, 0);
+            _rowFetchedCount = Math.Max((int)_rowFetchedCount, 0);
```

Comment:
I wonder if `nint.Max(_rowFetchedCount, 0)` would better match the intent or if explicitly truncating down to `int` was the intent...